### PR TITLE
Add safari extension

### PR DESCRIPTION
**Pull request by jaredhobbs**
Addresses #51 

Adds a popover toolbar button that loads the devtools.

screen shot 2017-06-13 at 17 10 32

### CODE REVIEW

**Comment by nickmessing** [comment]
Wow, good job

**Comment by yyx990803** [comment]
@jaredhobbs thanks a lot for the PR! I tried it locally (merged against latest `master`) but the popover is always blank, tried in both 10.1.1 and the Technology Preview - any ideas?

**Comment by jaredhobbs** [comment]
I'm on Safari 11 in the High Sierra beta. I developed it by running with `npm run dev:safari` then reloading the extension whenever the js rebuilt. The extension is set to only run on localhost and 127.0.0.1. You can open the web inspector in the popover by right clicking the popover window and clicking inspect element.

**Comment by yyx990803** [comment]
Can confirm it works when built in dev mode, however with `npm run build:safari` it shows up empty. Inspecting the popover doesn't seem to provide any useful info :(

**Comment by jaredhobbs** [comment]
:(

I'll keep on hacking to see what's going on in production mode.

**Comment by jaredhobbs** [comment]
ok, `npm run build:safari` works now

File: src/bridge.js [hunk 1]
```diff
@@ -3,13 +3,14 @@ import { EventEmitter } from 'events'
 export default class Bridge extends EventEmitter {
   constructor (wall) {
     super()
-    this.setMaxListeners(Infinity)
-    this.wall = wall
+    const self = this
```
Comment (posva) [inline src/bridge.js]: why did you add this?

Comment (jaredhobbs) [inline src/bridge.js]: that's what was causing the error in the safari build: `ReferenceError: Cannot access uninitialized variable.`

Might be related to https://bugs.webkit.org/show_bug.cgi?id=171543

Comment (posva) [inline src/bridge.js]: Ok, let's add a comment then 🙂

Comment (jaredhobbs) [inline src/bridge.js]: done

**Comment by yyx990803** [comment]
Thanks!

**Comment by thecrypticace** [comment]
Might be worth noting in the docs that you will get a blank popover (with console errors) when not running against 127.0.0.1 or localhost (unless you update the extension permissions in the builder).

Perhaps it would be good to trap the error for `safari.application.activeBrowserWindow.activeTab.page` being undefined and throw up a message?

**Comment by thecrypticace** [comment]
Also, holy mackerel this is awesome!

**Comment by tremendus** [comment]
@jaredhobbs This didn't work for me on Sierra with Safari 10 when running on localhost with `npm run build:safari`. Blank popover as reported by Evan.

Is 11 and/or High Sierra and/or dev mode a requirement or should this have worked on Sierra + 10 without dev?

**Comment by thecrypticace** [comment]
I'm running this on Sierra & Safari 10. Right click the popup and open the console. Do you see any errors?

**Comment by jaredhobbs** [comment]
Also make sure you're running the latest code from this repo and the site running at localhost is in development mode

**Comment by tremendus** [comment]
So interestingly I reloaded the page a bunch of times before I took of for dinner and it wasn't working, but I come back few hours later - no changes to anything - and it works just fine. 

Nevermind, and thanks.

**Comment by ruchern** [comment]
Hi Jared,

Can I just check and confirm that there is no way to add a tab to the Safari developer tools along side "elements, console, etc..."?

**Comment by jaredhobbs** [comment]
Unfortunately, no, at least as far as I could tell

**Comment by bigsee** [comment]
@jaredhobbs firstly, thank you for your awesome work here - so useful! 👍🏽

The extension seems to be working on a production site in Safari, even though I have disabled in my application. Same URL in Chrome gives the usual error about DevTools being disabled in production and does not display the devtools tab in the inspector.

Is this expected behaviour? i.e. will Safari devtools always work even against sites in production?

**Comment by jaredhobbs** [comment]
It shouldn't. If you had opened the tools on a dev site and then navigated to a prod site, the tools will still display info from the dev site. It's a known bug and should clear out but doesn't.

**Comment by bigsee** [comment]
@jaredhobbs thank you for the swift response. :) I can confirm this behaviour after testing so I'll leave open the issue I created here: https://github.com/vuejs/vue-devtools/issues/632
